starport_template/lib/widgets/sign_transaction_tab_view_item.dart

```diff
@@ -21,14 +22,23 @@ class SignTransactionTabViewItem extends StatelessWidget {
     final theme = CosmosTheme.of(context);
     return Padding(
       padding: EdgeInsets.symmetric(horizontal: theme.spacingL),
-      child: Row(
+      child: Column(
         mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        crossAxisAlignment: CrossAxisAlignment.start,
         children: [
-          Text(text, style: CosmosTextTheme.titleS),
-          const Spacer(),
-          Text('${amount.toString()} ${balance.denom.text.toUpperCase()}', style: CosmosTextTheme.title1Medium),
+          Row(
+            children: [
+              Text(text, style: CosmosTextTheme.titleS),
+              const Spacer(),
+              CosmosTokenAvatar(text: balance.denom.text),
+            ],
+          ),
+          SizedBox(width: theme.spacingL),
+          Text(
+            '${NumberFormat.decimalPercentPattern(decimalDigits: 5).format(amount)} ${balance.denom.text.toUpperCase()}',
```

Comment:
we're introducing yet another formatting of the amount. let's extract formatting of the amount to the`Balance` class

Reply:
formatting amount is rather easy task and can differ between emeris and template and other apps, so no need to put it in cosmos_utils. its enough to have a helper method inside either the `Balance` class or `Denom` class to format it, currently, in Emeris we use the following:

```dart
String amountWithDenomText(Amount amount) => '${formatAmount(amount.value.toDouble(), symbol: '')} $displayName';
```
inside the `Denom` class, and the `Balance class has the same method that proxies to `Denom`:

```dart
String get amountWithDenomText => denom.amountWithDenomText(amount);
```

Reply:
Great

Reply:
![image](https://user-images.githubusercontent.com/25801929/162921224-5ddbd1a0-ac23-4641-84e3-cd3024733340.png)
@andrzejchm this gives you this